services/viva/microservice/src/lambdas/copyAttachment.ts

```diff
@@ -0,0 +1,77 @@
+import { wrappers } from '../libs/lambdaWrapper';
+import * as dynamoDb from '../libs/dynamoDb';
+import config from '../libs/config';
+import S3 from '../libs/S3';
+
+import { CasePersonRole } from '../types/caseItem';
+import type { CaseItem } from '../types/caseItem';
+
+interface GetCasesResponse {
+  Items: CaseItem[];
+}
+
+interface S3EventObject {
+  key: string;
+}
+export interface EventDetail {
+  object: S3EventObject;
+}
+
+export interface Dependencies {
+  copyFile: (sourceKey: string, destinationKey: string) => Promise<void>;
+  getLatestUpdatedCase: (personalNumber: string) => Promise<CaseItem | undefined>;
+}
+
+export interface LambdaRequest {
+  detail: EventDetail;
+}
+
+function copyFile(sourceKey: string, targetKey: string): Promise<void> {
+  const BUCKET_NAME = process.env.BUCKET_NAME ?? '';
+
+  return S3.copyFileWithinBucket(BUCKET_NAME, sourceKey, targetKey);
+}
+
+async function getLatestUpdatedCase(personalNumber: string): Promise<CaseItem | undefined> {
+  const PK = `USER#${personalNumber}`;
+
+  const queryParams = {
+    TableName: config.cases.tableName,
+    KeyConditionExpression: 'PK = :pk',
+    ExpressionAttributeValues: {
+      ':pk': PK,
+    },
+    ProjectionExpression: 'persons, updatedAt',
+  };
+
+  const result = (await dynamoDb.call('query', queryParams)) as GetCasesResponse;
+  const latestCase = (result.Items ?? []).sort((a, b) => a.updatedAt - b.updatedAt);
+
+  return latestCase[0];
+}
+
+export async function copyAttachment(input: LambdaRequest, dependencies: Dependencies) {
+  const [personalNumber, filename] = input.detail.object.key.split('/');
```

Comment:
Vill man ha med hela den specifika datastrukturen från S3?
Tänker att splitten sker utanför lambdan?

Reply:
Önskvärt vore att bara skicka in det som funktionen behöver. Dock så har vi inte kommit överens om hur (eller vart?) vi ska göra det.